src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -263,6 +272,10 @@ public Optional<Rotation2d> getDesiredAngleToLock(Pose2d robotPose, Pose3d[] fie
     // Get the angle of 0,0 to the turret pose
     desiredLockingAngle = new Rotation2d(relativeToTarget.getX(), relativeToTarget.getY());
 
+    // Account for robot rotation
+    desiredLockingAngle = desiredLockingAngle
+        .rotateBy(robotPose.getRotation().minus(new Rotation2d().fromDegrees(180)));
```

Comment:
This "minus" is kinda weird and i really don't like it but it works. If anyone else can think or congure up a better way of doing this (I'm sure that there _is_ a better way, i just cant think of it right now), i am taking suggestions